src/System.IO.FileSystem.Watcher/tests/Utility/FileSystemWatcherTest.cs

```diff
@@ -491,5 +491,82 @@ private static FileSystemWatcher RecreateWatcher(FileSystemWatcher watcher)
 
             return newWatcher;
         }
+
+        internal readonly struct FiredEvent
+        {
+            public FiredEvent(WatcherChangeTypes eventType, string dir1, string dir2 = "") => (EventType, Dir1, Dir2) = (eventType, dir1, dir2);
+
+            public readonly WatcherChangeTypes EventType;
+            public readonly string Dir1;
+            public readonly string Dir2;
+
+            public override bool Equals(Object obj) => obj is FiredEvent ? this.Equals((FiredEvent)obj) : false;
+
+            public bool Equals(FiredEvent other) => this.EventType == other.EventType &&
+                this.Dir1 == other.Dir1 &&
+                this.Dir2 == other.Dir2;
+
+
+            public override int GetHashCode() => this.EventType.GetHashCode() ^ this.Dir1.GetHashCode() ^ this.Dir2.GetHashCode();
+
+            public override string ToString() => $"{EventType} {Dir1} {Dir2}";
+
+        }
+
+        // Observe until an expected count of events is triggered, otherwise fail. Return all collected events.
+        internal static List<FiredEvent> ExpectEvents(FileSystemWatcher watcher, int expectedEvents, Action action)
+        {
+            var eventsOccured = new AutoResetEvent(false);
```

Comment:
`AutoResetEvent` should be disposed.
```suggestion
            using var eventsOccured = new AutoResetEvent(false);
```